### PR TITLE
Sample store: Redis: add filters to get samples endpoints

### PR DESCRIPTION
**Pull request by pallavi2209**

### CODE REVIEW

File: api/v1/controllers/samples.js [hunk 1]
```diff
@@ -53,7 +53,9 @@ module.exports = {
   findSamples(req, res, next) {
```
Comment (iamigo) [inline api/v1/controllers/samples.js]: So theoretically we should be able to run *all* the api sample tests with the sampleStore feature flag enabled and everything should work?

Comment (pallavi2209) [inline api/v1/controllers/samples.js]: Yes, let me actually add all those here as well. More tests never hurt.

File: cache/models/samples.js [hunk 2]
```diff
@@ -47,92 +127,246 @@ function cleanAddAspectToSample(sampleObj, aspectObj, method) {
...
+    const sampAspArr = sampleName.split('|');
     const commands = [];
 
-    // get sample
+    if (sampAspArr.length < TWO) {
+      throw new redisErrors.ResourceNotFoundError({
+        explanation: 'Incorrect sample name.',
+      });
+    }
+
```
Comment (shriramshankar) [inline cache/models/samples.js]: looks like a validation that can be done in the API layer.

File: tests/cache/models/samples/get.js [hunk 3]
```diff
@@ -109,4 +243,73 @@ describe(`api::redisEnabled::GET ${path}`, () => {
...
+    .expect(constants.httpStatus.BAD_REQUEST)
+    .end((err, res) => {
+      if (err) {
+        return done(err);
+      }
+
+      return done();
+    });
+  });
+});
```
Comment (shriramshankar) [inline tests/cache/models/samples/get.js]: new line at EOF

File: cache/models/samples.js (hunk 2 above)
Comment (pallavi2209) [inline cache/models/samples.js]: I thought we are keeping only API related things in api layer, rest of the logic goes in redis layer. Plus, splitting in api layer would mean passing sample name and aspect name or split again in redis layer.

**Comment by pallavi2209** [comment]
@iamigo @shriramshankar updated the PR

File: cache/models/samples.js (hunk 2 above)
Comment (shriramshankar) [inline cache/models/samples.js]: I think we are moving the validations too. @iamigo Will your sample validation story take care of this in the API layer?

Comment (iamigo) [inline cache/models/samples.js]: The only validation piece I was going to be adding in the API layer is this: 
Check for uniqueness of relatedLinks names within a sample. Do this on sample POST/PATCH/PUT when relatedLinks not empty. Reject with validation error if a relatedLink name is reused within the same sample.

File: cache/models/samples.js [hunk 4]
```diff
@@ -47,92 +127,250 @@ function cleanAddAspectToSample(sampleObj, aspectObj, method) {
...
+  if (!opts.order) {
+    resArr = applyLimitAndOffset(opts, sampKeysArr);
+  }
+
+  // apply wildcard expr on name, if specified
+  if (opts.filter && opts.filter.name) {
+    const filteredKeys = filterByFieldWildCardExpr(
+      resArr, 'name', opts.filter.name
+    );
+    console.log(filteredKeys);
```
Comment (shriramshankar) [inline cache/models/samples.js]: console logs

File: cache/models/samples.js [hunk 5]
```diff
@@ -16,12 +16,92 @@ const u = require('../../api/v1/helpers/verbs/utils');
...
+    }
+
+    if (strA > strB) {
+      return isDescending ? MINUS_ONE : ONE;
+    }
+
+    return ZERO;
+  });
+}
+
```
Comment (shriramshankar) [inline cache/models/samples.js]: this is nicely done.

**Comment by pallavi2209** [comment]
@shriramshankar Removed console log. Updated the PR.

**Pull request merged**